src/test/java/teammates/test/cases/webapi/GetStudentProfileActionTest.java

```diff
@@ -28,80 +30,218 @@ protected String getRequestMethod() {
+        loginAsStudent(student2InCourse1.googleId);
+        testGetCorrectProfile(expectedProfile, expectedName);
+    }
+
+    private void testGetCorrectProfile(StudentProfileAttributes expectedProfile,
+                                       String expectedName, String... submissionParams) {
         GetStudentProfileAction action = getAction(submissionParams);
         JsonResult result = getJsonResult(action);
-
         assertEquals(HttpStatus.SC_OK, result.getStatusCode());
-        StudentProfile output = (StudentProfile) result.getOutput();
-        output.getStudentProfile().modifiedDate = null; // ignore this field for testing
-        assertEquals(studentProfileAttributes.toString(), output.getStudentProfile().toString());
-        assertEquals(student.getName(), output.getName());
+        StudentProfileData actualProfile = (StudentProfileData) result.getOutput();
+        assertEquals(expectedName, actualProfile.getName());
+        assertEquals(expectedProfile.email, actualProfile.getEmail());
+        assertEquals(expectedProfile.shortName, actualProfile.getShortName());
+        assertEquals(expectedProfile.institute, actualProfile.getInstitute());
+        assertEquals(expectedProfile.moreInfo, actualProfile.getMoreInfo());
+        assertEquals(expectedProfile.nationality, actualProfile.getNationality());
+        assertEquals(expectedProfile.gender, actualProfile.getGender());
+        assertEquals(expectedProfile.pictureKey, actualProfile.getPictureKey());
     }
 
-    private void testActionInMasquerade(AccountAttributes student) {
-        gaeSimulation.loginAsAdmin("admin.user");
+    @Test
+    @Override
+    protected void testAccessControl() throws Exception {
+        // See test cases below.
+    }
 
-        ______TS("Typical case: masquerade mode");
-        StudentProfileAttributes student1InCourse1Profile = typicalBundle.profiles.get("student1InCourse1");
+    @Test
+    public void testAccessControl_withoutCorrectAuthInfo_shouldFail() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+        };
+        verifyInaccessibleWithoutLogin(submissionParams);
+        verifyInaccessibleForUnregisteredUsers(submissionParams);
+    }
 
+    @Test
+    public void testAccessControl_studentNotExistInCourse_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        StudentAttributes student1InCourse2 = typicalBundle.students.get("student1InCourse2");
+        CourseAttributes typicalCourse1 = typicalBundle.courses.get("typicalCourse1");
+        loginAsInstructor(instructor1OfCourse1.googleId);
```

Comment:
Well, I think this is unnecessary because `execute()` is blind to who request the profile. All it does is to return the profile of the given google id, and we just need to make sure the function is called in integration test. Therefore I think testing these cases in `checkAccessControl()` is enough.

Reply:
I think it's unnecessary, because `studentNotFound` logic is blind to whether an instructor or student requests  the profile.